f5_openstack_agent/lbaasv2/drivers/bigip/icontrol_driver.py

```diff
@@ -1148,11 +1148,13 @@ def get_all_deployed_listeners(self, expand_subcollections=False):
     def purge_orphaned_nodes(self, tenant_members):
         node_helper = resource_helper.BigIPResourceHelper(
             resource_helper.ResourceType.node)
+        node_dict = dict()
         for bigip in self.get_all_bigips():
             for tenant_id, members in tenant_members.iteritems():
                 partition = self.service_adapter.prefix + tenant_id
                 nodes = node_helper.get_resources(bigip, partition=partition)
-                node_dict = {n.name: n for n in nodes}
+                for n in nodes:
+                    node_dict[n.name] = n
```

Comment:
I just tried in simple test: 

```
class Node:
    def __init__(self, n):
        self.name = "%d" % n

if __name__ == "__main__":
    nodes = [Node(1), Node(2), Node(3)]
    node_dict = {n.name: n for n in nodes}
    print(node_dict)
```
It reports no error..  why it happens in this code?

Reply:
The dict comprehension only works for Python 2.7+, but we use 2.6 to run the CI. It better to choose the code for backward compatibility.